java/fury-core/src/main/java/org/apache/fury/resolver/UserContextResolver.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.fury.resolver;
+
+import org.apache.fury.Fury;
+import org.apache.fury.memory.MemoryBuffer;
+
+/** write/read user custom global data after metaContext data. */
+public abstract class UserContextResolver {
+
+  protected final Fury fury;
+
+  public UserContextResolver(Fury fury) {
+    this.fury = fury;
+  }
+
+  public abstract void write(MemoryBuffer buffer);
+
+  public abstract void read(MemoryBuffer buffer);
+
+  /** write/read end should clear user data. */
+  public abstract void reset();
```

Comment:
In meta share mode, we can write this context in a channel only once. If we reset it every time, does it means that we will write this every time?

Reply:
I would like context used without state, context is global in a fury and reset every write/read. if we do it like meta context, we need store contexts and fury. so I think context design should be same and I wander your advice.

Reply:
If it's stateless, we can implement this like :
```java
  @Override
  public void write(MemoryBuffer buffer, Object value) {
    Map<String, Integer> o = (Map<String, Integer>) fury.getSerializationContext().get("dict");
    if (o == null) {
      o = xxx;
      fury.getSerializationContext().add("dict", o);
      fury.writeNonRef(buffer, o);
      buffer.writeBoolean(true);
    } else {
      buffer.writeBoolean(false);
    }
    
  }

  @Override
  public Object read(MemoryBuffer buffer) {
    Map<String, Integer> o;
    if (buffer.readBoolean()) {
      o = (Map<String, Integer>) fury.readNonRef(buffer);
      fury.getSerializationContext().add("dict", o);
    } else {
      o = (Map<String, Integer>) fury.getSerializationContext().get("dict");
    }
  }
```

But it introduce one extra byte

Reply:
with `UserContext#reset()` just like `SerializationContext#reset()`, take user a choice  whether clean their data after fury serialize/deserialize. we can save tmp data in userContext and not in `SerializationContext#objects`.